pkg/build/pipeline/CiTemplateReadService.go

```diff
@@ -0,0 +1,184 @@
+package pipeline
+
+import (
+	"github.com/devtron-labs/devtron/internal/sql/repository/pipelineConfig"
+	"github.com/devtron-labs/devtron/internal/util"
+	"github.com/devtron-labs/devtron/pkg/build/pipeline/bean"
+	"github.com/devtron-labs/devtron/pkg/pipeline/adapter"
+	"github.com/devtron-labs/devtron/pkg/pipeline/bean/CiPipeline"
+	"github.com/devtron-labs/devtron/pkg/pipeline/types"
+	"github.com/go-pg/pg"
+	"go.uber.org/zap"
+)
+
+type CiTemplateReadService interface {
+	FindByAppId(appId int) (*bean.CiTemplateBean, error)
+	FindTemplateOverrideByAppId(appId int) ([]*bean.CiTemplateBean, error)
+	FindTemplateOverrideByCiPipelineIds(ciPipelineIds []int) ([]*bean.CiTemplateBean, error)
+	FindTemplateOverrideByCiPipelineId(ciPipelineId int) (*bean.CiTemplateBean, error)
+	GetAppliedDockerConfigForCiPipeline(ciPipelineId, appId int, isOverridden bool) (*types.DockerArtifactStoreBean, error)
+	GetBaseDockerConfigForCiPipeline(appId int) (*types.DockerArtifactStoreBean, error)
+	FindByAppIds(appIds []int) (map[int]*bean.CiTemplateBean, error)
+}
+
+type CiTemplateReadServiceImpl struct {
+	Logger                       *zap.SugaredLogger
+	CiTemplateRepository         pipelineConfig.CiTemplateRepository
+	CiTemplateOverrideRepository pipelineConfig.CiTemplateOverrideRepository
+}
+
+func NewCiTemplateReadServiceImpl(logger *zap.SugaredLogger, ciTemplateRepository pipelineConfig.CiTemplateRepository, ciTemplateOverrideRepository pipelineConfig.CiTemplateOverrideRepository) *CiTemplateReadServiceImpl {
+	return &CiTemplateReadServiceImpl{
+		Logger:                       logger,
+		CiTemplateRepository:         ciTemplateRepository,
+		CiTemplateOverrideRepository: ciTemplateOverrideRepository,
+	}
+}
+
+func (impl CiTemplateReadServiceImpl) FindByAppId(appId int) (ciTemplateBean *bean.CiTemplateBean, err error) {
```

Comment:
let us replace (impl CiTemplateReadServiceImpl) -> (impl *CiTemplateReadServiceImpl) in all methods and adapt this in all upcoming changes